src/EmailServiceClient.php

```diff
@@ -0,0 +1,260 @@
+    {
+
+        if (isset($config[self::ASSERT_VALID_IP_CONFIG])) {
+            $this->assertValidIp = $config[self::ASSERT_VALID_IP_CONFIG];
+        }
+
+        // If we don't need to validate the service Ip, we're done here
+        if ( ! $this->assertValidIp) {
+            return;
+        }
+
+        /*
+         *  If we should validate the service IP but there aren't
+         *  any trusted IPs, throw an exception
+         */
+        if (empty($config[self::TRUSTED_IPS_CONFIG])) {
+            throw new \InvalidArgumentException(
+                'The config entry for ' . self::TRUSTED_IPS_CONFIG .
+                ' must be set (as an array) when ' .
+                self::ASSERT_VALID_IP_CONFIG .
+                ' is not set or is set to True.',
+                1494531150
+            );
+        }
+
+        /*
+         * At this point, we need to validate the service Ip and we know
+         * that the TRUSTED_IPS_CONFIG is not empty
+         */
+        $newTrustedIpRanges = $config[self::TRUSTED_IPS_CONFIG];
+        if ( ! is_array($newTrustedIpRanges)) {
+            throw new \InvalidArgumentException(
+                'The config entry for ' . self::TRUSTED_IPS_CONFIG .
+                ' must be an array.',
+                1494531200
+            );
+        }
+
+        foreach ($newTrustedIpRanges as $nextIpRange) {
+            $ipBlock = IPBlock::create($nextIpRange);
+            $this->trustedIpRanges[] = $ipBlock;
+        }
+
+        $this->assertTrustedIp();
+    }
+
+    /**
+     * Attempt to authenticate using the given credentials, getting back
+     * information about the authenticated user (if the credentials were
+     * acceptable) or null (if unacceptable).
+     *
+     * @param string $username The username.
+     * @param string $password The password (in plaintext).
+     * @return array|null An array of user information (if valid), or null.
+     * @throws Exception
+     */
+    public function authenticate(string $username, string $password)
+    {
+        $result = $this->authenticateInternal([
+            'username' => $username,
+            'password' => $password,
+        ]);
+        $statusCode = (int)$result['statusCode'];
+        
+        if ($statusCode === 200) {
+            return $this->getResultAsArrayWithoutStatusCode($result);
+        } elseif ($statusCode === 400) {
+            return null;
+        }
+        
+        $this->reportUnexpectedResponse($result, 1490802360);
+    }
+    
+    /**
+     * Create a user with the given information.
+     *
+     * @param array $config An array key/value pairs of attributes for the new
+     *     user.
+     * @return array An array of information about the new user.
+     * @throws Exception
+     */
+    public function email(array $config = [])
+    {
+        $result = $this->emailInternal($config);
+        $statusCode = (int)$result['statusCode'];
+        
+        if ($statusCode === 200) {
```

Comment:
Do we want to allow any `2xx` response? We had to do that for siteStatus (see [line 204](https://github.com/silinternational/email-service-php-client/pull/1/files#diff-32b9ee4d143c66b3fa426dd636c436d8R204), below).